### PR TITLE
Add English WNP for Firefox 107 [fix #12229]

### LINKED ISSUE #12229: WNP 107 (en)
## Description

Design: [here](https://www.figma.com/file/CZvzHgU9NQMQUm34lomIl5/Firefox-WNP?node-id=2478%3A39707)

Copy: [here](https://docs.google.com/document/d/1gtie8VFq0uDJgffyZppkv-1GwJCS3MqLKKpoVYWAM60/edit?usp=sharing)

### Success Criteria:
- [ ] Switch to easily remove coupon code section
- [ ] helper is used for attribution

### PR DESCRIPTION
**Pull request by craigcook**
## One-line summary
Adds the English what's new page for Firefox 107 (I'm not sure if the EU team wants a different page shown in the UK, but I'm assuming for now this one is for all Englishes and not just US and CA). @maureenlholland is working on the EU pages.

## Issue / Bugzilla link
#12229 

## Testing
http://localhost:8000/en-US/firefox/107.0/whatsnew/

### CODE REVIEW

**Comment by craigcook** [comment]
Looks like the coupon version is part of an experiment (I missed that detail) so I need to change some things.

**Comment by craigcook** [comment]
OK, it's not an experiment after all. Everybody gets a coupon code! 🎅

**Comment by maureenlholland** [comment]
I've only got DE and FR pages for the EU version, so I assume UK is meant to go under the EN design in this case 👍

File: bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx107-en.html [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  <h3 class="wnp-sub-title">Get Mozilla VPN</h3>
+
+  <p class="wnp-main-tagline">
+    With travel logistics, holiday parties and gift ideas — you have too much on
+    your plate to worry about privacy. Mozilla VPN protects your location and
+    browsing activity when you’re online, so you can have peace of mind.
+  </p>
+
+  <p class="wnp-main-cta">
+    {{ vpn_product_referral_link(
```
Comment (maureenlholland) [inline bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx107-en.html:34]: ✅

File: bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx107-en.html [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+          'data-cta-type' : 'button'
+      })
+    }}
+  </p>
+
+  <aside class="c-coupon">
+    <h3>Use code at checkout for 20% off</h3>
+    <p>
+      <strong class="c-coupon-code">VPNHOLIDAY20</strong>
+      <button type="button" class="c-coupon-copy" id="code-copy" data-code="VPNHOLIDAY20" data-success="Code copied!">Copy code</button>
```
Comment (maureenlholland) [inline bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx107-en.html:49]: nice

File: media/css/firefox/whatsnew/whatsnew-107-en.scss [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    .c-coupon-code {
+        @include text-body-xl;
+        border-radius: $border-radius-sm;
+        border: 1px solid $color-black;
+        display: inline-block;
+        font-weight: normal;
+        padding: $spacing-sm $spacing-md;
+    }
+
+    .c-coupon-copy {
```
Comment (maureenlholland) [inline media/css/firefox/whatsnew/whatsnew-107-en.scss:78]: **nit (non-blocking)**: top vertical spacing feels a bit tight, touch space is good horizontally but could be a little more generous vertically

copy-spacing

**suggestion**: add vertical padding (or at least top to space out the Copy Code text a bit more)

File: bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx107-en.html (hunk 2 above)
Comment (maureenlholland) [reply bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx107-en.html:49]:
```suggestion
      <button type="button" data-cta-type="button" data-cta-text="Copy code" class="c-coupon-copy" id="code-copy" data-code="VPNHOLIDAY20" data-success="Code copied!">Copy code</button>
```

non-blocking, not sure if we're logging this anywhere else, but maybe worth having some info on whether anyone's using this functionality

**Review by maureenlholland — APPROVED** [review]
LGTM in light and dark mode 🌞 🌔 

in the issue there's a request for a switch to be able to quickly remove the coupon section, is that still required?

non-blocking nitty stuff in comments

**Comment by craigcook** [comment]
Updated!

File: bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx107-en.html [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      link_text='Get Mozilla VPN',
+      class_name='mzp-t-product',
+      optional_attributes= {
+          'data-cta-text' : 'Get Mozilla VPN',
+          'data-cta-type' : 'button'
+      })
+    }}
+  </p>
+
+  {% if switch('firefox_whatsnew_107_en_coupon') %}
```
Comment (maureenlholland) [inline bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx107-en.html:45]: that's interesting, I thought these needed kebab casing, but tested locally, it works as expected 💯

**Review by maureenlholland — APPROVED** [review]
r++

**Comment by maureenlholland** [comment]
before merge, need to add to list here: https://github.com/mozilla/bedrock/blob/main/bedrock/firefox/views.py#L499

File: bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx107-en.html (hunk 4 above)
Comment (craigcook) [reply bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx107-en.html:45]: Either works, I can never remember which is the standard though... I guessed wrong this time.

**Comment by craigcook** [comment]
> before merge, need to add to list here: https://github.com/mozilla/bedrock/blob/main/bedrock/firefox/views.py#L499

Already did!

**Pull request merged**